### PR TITLE
update-SSO-troubleshooting docs

### PR DESCRIPTION
**Pull request by oshati**
PR to update the SSO Troubleshooting FAQ based on the below comment
https://github.com/gravitational/customer-sensitive-requests/issues/120#issuecomment-1706949583

### CODE REVIEW

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx [hunk 1]
```diff
@@ -58,3 +58,12 @@ spec:
 version: v5
 ```
 
+###Failed to verify JWT: oidc: unable to verify JWT signature: no matching keys
```
Comment (zmb3) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:61]:
```suggestion
### Single sign-on fails fails with OIDC

When using OIDC to authenticate with Teleport, you may see failed
login attempts with the following error logs:

```
Failed to verify JWT: oidc: unable to verify JWT signature: no matching keys
```

```

Add a space before the title and rephrase it to describe the problem.

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx [hunk 2]
```diff
@@ -58,3 +58,12 @@ spec:
...
+
+The error typically indicates a discrepancy between the algorithm used to sign the JWT token and the algorithm(s) supported by the JWKS (JSON Web Key Set). Specifically, the token might be signed with one algorithm, e.g., HS256, while the JWKS only lists keys for a different algorithm. e.g., RS256. This issue predominantly arises when using identity providers that offer extremely low-level functionality.
```
Comment (zmb3) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:63]:
```suggestion
The error typically indicates a discrepancy between the algorithm used to sign the JWT token and the algorithm(s) supported by the JSON Web Key Set (JWKS). Specifically, the token might be signed with one algorithm, e.g., HS256, while the JWKS only lists keys for a different algorithm. e.g., RS256. This issue predominantly arises when using identity providers that offer extremely low-level functionality.
```

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx [hunk 3]
```diff
@@ -58,3 +58,12 @@ spec:
...
+
+Here are some things to check:
+- Verify the JWT header specifies the correct signing algorithm (alg). This should match one of the algorithms listed in the keys section of the JWKS endpoint response. Common algorithms are RS256.
```
Comment (zmb3) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:66]: Should we be more explicit here? Does Teleport only support RS256?

**Review by zmb3 — COMMENTED** [review]
Thank you for following up on this.

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx (hunk 3 above)
Comment (ptgott) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:66]:
```suggestion
- Verify the JWT header specifies the correct signing algorithm. This should match one of the algorithms listed in the keys section of the JWKS endpoint response. Common algorithms are RS256.
```
We don't mention the abbreviation again outside of parentheses, so I think it's fine to remove it.

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx [hunk 4]
```diff
@@ -58,3 +58,12 @@ spec:
...
+- Ensure the JWKS endpoint is returning all relevant public keys. Sometimes key rotation can cause valid keys to be omitted.
+
+To resolve the issue, align the JWT algorithm header (alg) with a supported algorithm in the JWKS. Rotate keys if necessary. Verify the JWKS only publishes the active public keys. With proper configuration, the signature should validate successfully.
```
Comment (ptgott) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:69]:
```suggestion
To resolve the issue, align the JWT algorithm header with a supported algorithm in the JWKS. Rotate keys if necessary. Verify the JWKS only publishes the active public keys. With proper configuration, the signature should validate successfully.
```

**Review by ptgott — APPROVED** [review]

**Comment by ptgott** [comment]
@oshati Added backport labels. Feel free to remove any that aren't appropriate!

File: docs/pages/includes/sso/loginerrortroubleshooting.mdx [hunk 5]
```diff
@@ -58,3 +58,12 @@ spec:
 version: v5
 ```
 
+### Single sign-on fails with OIDC
+
+The error typically indicates a discrepancy between the algorithm used to sign the JWT token and the algorithm(s) supported by the JSON Web Key Set (JWKS). Specifically, the token might be signed with one algorithm, e.g., HS256, while the JWKS only lists keys for a different algorithm. e.g., RS256. This issue predominantly arises when using identity providers that offer extremely low-level functionality.
```
Comment (zmb3) [inline docs/pages/includes/sso/loginerrortroubleshooting.mdx:63]: I think we should still mention the actual error that the user will see in this case (Failed to verify JWT: oidc: unable to verify JWT signature: no matching keys), my suggestion was just that we don't make it the title of the section.

**Review by zmb3 — APPROVED** [review]
